### PR TITLE
SONARJAVA-4947 Add SingleFileVerifier.assertNoIssuesRaised

### PR DESCRIPTION
**Pull request by leonardo-pilastri-sonarsource**

### CODE REVIEW

File: test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/SingleFileVerifier.java [hunk 1]
```diff
@@ -99,6 +99,11 @@ static SingleFileVerifier create(Path sourceFilePath, Charset encoding) {
    */
   void assertNoIssues();
 
+  /**
+   * Run the comparison and expect to find no issue. Allows Noncompliant comments to be present in the file.
```
Comment (johann-beleites-sonarsource) [inline test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/SingleFileVerifier.java:103]: Let's also update the comment of `assertNoIssues()` to explicitly mention that noncompliant comments are not allowed.

File: test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/internal/InternalIssueVerifier.java [hunk 2]
```diff
@@ -135,6 +135,12 @@ public void assertOneOrMoreIssues() {
     assertIssues(true);
   }
 
+  @Override
+  public void assertNoIssuesRaised() {
+    Report report = buildReport();
+    assertEquals("ERROR: Found " + report.getActualIssueCount() + " unexpected issues.", 0, report.getActualIssueCount());
```
Comment (johann-beleites-sonarsource) [inline test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/internal/InternalIssueVerifier.java:141]: Can we add an explicit message about the fact that we expect no issues at all? My fear would be that someone uses the method, then there are issues raised, and they look into the sample file, assuming we called `verifyIssues()`, having a hard time figuring out that it is the expectation that is different to theirs.

**Review by johann-beleites-sonarsource — COMMENTED** [review]
I have a couple of suggestions just for clarity of the messaging for users of the library, who may not always be super familiar with it.

File: test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/internal/InternalIssueVerifier.java (hunk 2 above)
Comment (leonardo-pilastri-sonarsource) [reply test-commons/src/main/java/org/sonarsource/analyzer/commons/checks/verifier/internal/InternalIssueVerifier.java:141]: What about "ERROR: No issues were expected, but some were found."?

**Review by johann-beleites-sonarsource — APPROVED** [review]

**Pull request merged**